backend/src/error.rs

```diff
@@ -0,0 +1,35 @@
+#![allow(clippy::module_name_repetitions)]
+use std::fmt;
+
+use axum::response::{IntoResponse, Response};
+use error_stack::Context;
+use hyper::StatusCode;
+
+/// Server start up errors
+#[derive(Debug)]
+pub enum AppError {
+    InitializationError,
+    StartUpError,
+}
+
+impl fmt::Display for AppError {
```

Comment:
I think it is better to use `thiserror` crate for `AppError` enum. It allows to write specific error message as an attribute above each enum variant, so you don't need to implement `Display` manually. Additionally, this crate will add `Error` trait support automatically